src/services/PlateGenerationService.ts

```diff
@@ -0,0 +1,209 @@
+                    fileFormat: "pdf",
+                    fileName: `${vehicle.vin}_${techRecord.plates.plateSerialNumber}.pdf`,
+                    fileSize: responseBuffer.byteLength.toString(),
+                    certificate: responseBuffer,
+                    email: techRecord.plates.toEmailAddress
+                };
+            })
+            .catch((error: AWSError | Error) => {
+                console.log(error);
+                throw error;
+            });
+
+    }
+
+    /**
+     * Generates the payload for the MOT certificate generation service
+     * @param vehicle - source tech record for plate generation
+     */
+    public async generatePayload(vehicle: ITechRecordWrapper) {
+        let payload: ICertificatePayload = {
+            Watermark: (process.env.BRANCH === "prod") ? "" : "NOT VALID",
+            PLATES_DATA: undefined
+        };
+        const platesData = this.generateCertificateData(vehicle);
+        payload.PLATES_DATA = platesData;
+
+        // Purge undefined values
+        payload = JSON.parse(JSON.stringify(payload));
+        return payload;
+    }
+
+    /**
+     * Generates plates data for a given vehicle
+     * @param vehicle - the source vehicle for plates generation
+     */
+    public generateCertificateData(vehicle: ITechRecordWrapper) {
+        const techRecord: ITechRecord = vehicle.techRecord[0];
+        const platePayload = {
+            PlateSerialNumber: techRecord.plates.plateSerialNumber,
+            DtpNumber: techRecord.brakes.dtpNumber,
+            PrimaryVrm: vehicle.primaryVrm,
+            Vin: vehicle.vin,
+            VariantNumber: techRecord.variantNumber,
+            ApprovalTypeNumber: techRecord.approvalTypeNumber,
+            Make: techRecord.make,
+            Model: techRecord.model,
+            SpeedLimiterMrk: techRecord.vehicleType === VEHICLE_TYPES.HGV ? this.getSpeedLimiterMrk(techRecord.speedLimiterMrk) : undefined,
+            FunctionCode: techRecord.functionCode,
+            RegnDate: techRecord.regnDate,
+            ManufactureYear: this.getFormattedField(techRecord.manufactureYear),
+            GrossGbWeight: this.getFormattedField(techRecord.grossGbWeight),
+            GrossEecWeight: this.getFormattedField(techRecord.grossEecWeight),
+            GrossDesignWeight: this.getFormattedField(techRecord.grossDesignWeight),
+            TrainGbWeight: this.getFormattedField(techRecord.trainGbWeight),
+            TrainEecWeight: this.getFormattedField(techRecord.trainEecWeight),
+            TrainDesignWeight: this.getFormattedField(techRecord.trainDesignWeight),
+            MaxTrainGbWeight: this.getFormattedField(techRecord.maxTrainGbWeight),
+            MaxTrainEecWeight: this.getFormattedField(techRecord.maxTrainEecWeight),
+            Axles: this.getAxles(techRecord),
+            MaxLoadOnCoupling: techRecord.vehicleType === VEHICLE_TYPES.TRL ? this.getFormattedField(techRecord.maxLoadOnCoupling) : undefined,
```

Comment:
suggest to use short circuit approach e.g.
`techRecord.vehicleType === VEHICLE_TYPES.TRL && this.getFormattedField(techRecord.maxLoadOnCoupling)`